src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -913,15 +903,21 @@ public without sharing class BDI_MigrationMappingUtility {
                 }
             }
 
-            if (hasDataImportNames == true) {
-                Integer countWhitespaces = helpText.countMatches(' ');
```

Comment:
I see you removed countMatches() from here but it's still being used on line 896 and 898. Do we still need it there when parsing out the data import object names from the special characters?

Reply:
This is pre-existing code, but I don't quite follow the reason for this line `if (parts.size() >= 2 && countWhitespaces <= 4)`. [The `split` line above set a limit to 2, and I don't get the reason for the countWhitespaces]

Reply:
I'll let @psadie-sfdo respond definitively, but I think the criteria here were part of a *somewhat* arbitrary set of metrics hashed out through an extensive set of conversations, where the end goal is to attempt to identify the distinction between "help text that was intended to be a field mapping" and "help text that was standard help text not intended to be a field mapping". 

The first part: If parts.size() == 2, then we know the help text contained a special character that MIGHT have been intended to split the object name and the field name, which is supposed to be '.' but we gave a lot of latitude for other possible delimiters. 

The second part: I think "4" is very arbitrary, but the idea is that more than a certain threshold of spaces probably indicated an intent to set human-readable help text in the form of a sentence, but fewer than that threshold (along with a delimiting character) might indicate a misguided use of a field label like "Opportunity Close Date" instead of the API name, which of course might have no spaces.  

Honestly, at the time, and to this day still, I think we went too far in the direction of charitably attempting to view something as an attempt at help text mapping, although I suppose something like "Opportunity$Opportunity Close Date" is as likely as not to be an attempted mapping? I wouldn't mind tightening up this criteria a bit, just to avoid inadvertent interpretations of "normal" help text as an attempted mapping gone awry, which I think is basically what introduced the bug we are fixing here.

Reply:
@psadie-sfdo - This code might benefit from some comments explaining some of the thinking in more detail, regardless

Reply:
Patrick's spot on. I'm taking another pass at this. Add comments and _possibly_ tighten up the criteria if it feels safe. Completely agree that this was trying to be overly helpful. I'll ping for a re-review when that's ready. Thanks folks!